services/api/pepc_boost_test.go

```diff
@@ -226,813 +318,718 @@ func TestCheckTxAndSenderValidity(t *testing.T) {
-	require.Equal(t, tobTxValue, big.NewInt(2))
+	for _, c := range cases {
+		t.Run(c.description, func(t *testing.T) {
 
-	tobtxs, err := backend.redis.GetTobTx(context.Background(), backend.redis.NewTxPipeline(), headSlot+1, parentHash)
-	require.NoError(t, err)
+			backend := newTestBackend(t, 1)
+
+			headSlot := uint64(32)
+			parentHash := "0xbd3291854dc822b7ec585925cda0e18f06af28fa2886e15f52d52dd4b6f94ed6"
+			feeRec, err := types.HexToAddress("0x5cc0dde14e7256340cc820415a6022a7d1c93a35")
+			require.NoError(t, err)
+			withdrawalsRoot, err := hexutil.Decode("0xb15ed76298ff84a586b1d875df08b6676c98dfe9c7cd73fab88450348d8e70c8")
+			require.NoError(t, err)
+			prevRandao := "0x9962816e9d0a39fd4c80935338a741dc916d1545694e41eb5a505e1a3098f9e4"
+			proposerPubkeyByte, err := hexutil.Decode(testProposerKey)
+			require.NoError(t, err)
+			proposerPubkey := phase0.BLSPubKey(proposerPubkeyByte)
+
+			prepareBackend(backend, headSlot, parentHash, feeRec, withdrawalsRoot, prevRandao, proposerPubkey)
+
+			// submit first set of tob txs
+			tobTxReqs := bellatrixUtil.ExecutionPayloadTransactions{Transactions: []bellatrix.Transaction{}}
+			for _, tx := range c.firstTobTxs {
+				txByte, err := tx.MarshalBinary()
+				require.NoError(t, err)
+				tobTxReqs.Transactions = append(tobTxReqs.Transactions, txByte)
+			}
+			firstSetTxHashRoot, err := tobTxReqs.HashTreeRoot()
+			require.NoError(t, err)
+			req := &common.TobTxsSubmitRequest{
+				ParentHash: parentHash,
+				TobTxs:     tobTxReqs,
+				Slot:       headSlot + 1,
+			}
+			jsonReq, err := req.MarshalJSON()
+			require.NoError(t, err)
+			rr := backend.requestBytes(http.MethodPost, tobTxSubmitPath, jsonReq, map[string]string{
+				"Content-Type": "application/json",
+			})
+			require.NoError(t, err)
+			require.Equal(t, http.StatusOK, rr.Code)
+			// first checks should check for the first set of tob txs
+			assertTobTxs(t, backend, headSlot+1, parentHash, c.firstTobTxs[len(c.firstTobTxs)-1].Value(), firstSetTxHashRoot)
+
+			// submit second set of txs
+			tobTxReqs = bellatrixUtil.ExecutionPayloadTransactions{Transactions: []bellatrix.Transaction{}}
+			for _, tx := range c.secondTobTxs {
+				txByte, err := tx.MarshalBinary()
+				require.NoError(t, err)
+				tobTxReqs.Transactions = append(tobTxReqs.Transactions, txByte)
+			}
+			secondSetTxHashRoot, err := tobTxReqs.HashTreeRoot()
+			require.NoError(t, err)
+			req = &common.TobTxsSubmitRequest{
+				ParentHash: parentHash,
+				TobTxs:     tobTxReqs,
+				Slot:       headSlot + 1,
+			}
+			jsonReq, err = req.MarshalJSON()
+			require.NoError(t, err)
+			rr = backend.requestBytes(http.MethodPost, tobTxSubmitPath, jsonReq, map[string]string{
+				"Content-Type": "application/json",
+			})
+			if !c.nextSentIsHigher {
+				require.NoError(t, err)
+				require.Equal(t, http.StatusBadRequest, rr.Code)
+				require.Contains(t, rr.Body.String(), "TOB tx value is less than the current value!")
+			} else {
```

Comment:
add return to if and then no need for the else

Reply:
got it!